src/jit.rs

```diff
@@ -248,71 +247,45 @@ fn emit_pop(jit: &mut JitMemory, r: u8) {
 // We use the MR encoding when there is a choice
 // 'src' is often used as an opcode extension
 #[inline]
-fn emit_alu32(jit: &mut JitMemory, op: u8, src: u8, dst: u8) {
-    emit_basic_rex(jit, 0, src, dst);
+fn emit_alu(jit: &mut JitMemory, w: u8, op: u8, src: u8, dst: u8, imm: i32, displacement: Option<i32>) {
```

Comment:
Nit: The `w` here is a bit confusin if you can't tell from the prototype that it refers to the REX.w.  Maybe call it `rex_w` here (but keep it as just `w` in the rex specific calls above